src/Screens/Cart/Cart.js

```diff
@@ -0,0 +1,223 @@
+import { useEffect } from "react";
+import { useProducts } from "./../../Context/products-context";
+import { Link } from "react-router-dom";
+import Empty from "./../../Components/Empty/Empty";
+import emptyImage from "./../../Media/empty-cart.png";
+import { useToast } from "./../../Context/toast-context";
+import Navbar from "./../../Components/Navbar/Navbar";
+
+function Cart() {
+  const encodedToken = localStorage.getItem("ENCODED_TOKEN");
+
+  const { toggleToast, toastVisibility, toastColor, toastText } = useToast();
+
+  const { cartArray, setCartArray, wishlistArray, setWishlistArray } =
+    useProducts();
+
+  const removeFromCart = (id) => {
+    setCartArray(cartArray.filter((cartItem) => cartItem._id !== id));
+    toggleToast("Removed From Cart ✔", "red", "whitesmoke");
+
+    fetch(`/api/user/cart/${id}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        authorization: encodedToken,
+      },
+    })
+      .then((res) => res.json())
+      .then((data) => console.log(data));
+  };
+
+  const moveToWishlist = (product) => {
+    var wishlistFlag = false;
+    wishlistArray.map((wishlistItem) => {
+      if (wishlistItem._id === product._id) {
+        wishlistFlag = true;
+        return true;
+      }
+      return true;
+    });
+    if (wishlistFlag === false) {
+      setWishlistArray([...wishlistArray, product]);
+    }
+    toggleToast("Moved To Wishlist ✔", "green", "whitesmoke");
+    setCartArray(cartArray.filter((cartItem) => cartItem._id !== product._id));
+
+    fetch(`/api/user/cart/${product._id}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        authorization: encodedToken,
+      },
+    })
+      .then((res) => res.json())
+      .then((data) => console.log(data))
+      .then(() => {
+        fetch("/api/user/wishlist", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            authorization: encodedToken,
+          },
+          body: JSON.stringify({ product }),
+        })
+          .then((res) => res.json())
+          .then((data) => console.log(data));
+      });
+  };
+
+  const incrementCartItemQuantity = (id) => {
+    cartArray.map((cartItem, index) => {
+      if (cartItem._id === id) {
+        setCartArray([
+          ...cartArray.slice(0, index),
+          { ...cartArray[index], bookQuantity: cartItem.bookQuantity + 1 },
+          ...cartArray.slice(index + 1),
+        ]);
+      }
+      return true;
+    });
+    fetch(`/api/user/cart/${id}`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        authorization: encodedToken,
+      },
+      body: JSON.stringify({ action: { type: "increment" } }),
+    })
+      .then((res) => res.json())
+      .then((data) => console.log(data));
```

Comment:
I am not sure why you are console logging it.
For me, it's a little difficult to get the context of why it is done in case a fetch call is not being used we can drop it.

Reply:
I meant you can drop the step of console logging. just realized it is there for an incremental purpose.